courier/courier_dispatcher.go

```diff
@@ -48,10 +47,10 @@ func (c *courier) DispatchQueue(ctx context.Context) error {
 		return err
 	}
 
-	ttl := c.deps.CourierConfig(ctx).CourierMessageTTL()
+	maxRetries := c.deps.CourierConfig(ctx).CourierMessageRetries()
 
 	for k, msg := range messages {
-		if time.Now().After(msg.CreatedAt.Add(ttl)) {
+		if msg.SendCount >= maxRetries {
```

Comment:
I think we should check whether we exceed the max retries, not whether we reach them. That So if we say: please retry 5 times, it should retry 5 times and fail at the sixth time.

```suggestion
		if msg.SendCount > maxRetries {
```

Reply:
Right, done. A bit of an awkward spot to wrap the head around, so I added some comments in the test case explaining the behavior.